webapp/src/pages/registration/components/RegistrationComponent.js

```diff
@@ -0,0 +1,381 @@
+import { withRouter } from "react-router";
+
+import FormTextArea from "../../../components/form/FormTextArea";
+import FormTextBox from "../../../components/form/FormTextBox";
+import FormSelect from "../../../components/form/FormSelect";
+import FormCheckbox from "../../../components/form/FormCheckbox";
+import FormFileUpload from "../../../components/form/FormFileUpload";
+import { registrationService } from "../../../services/registration";
+import { fileService } from "../../../services/file/file.service";
+
+import { ConfirmModal } from "react-bootstrap4-modal";
+
+const DEFAULT_EVENT_ID = process.env.REACT_APP_DEFAULT_EVENT_ID || 1;
+const baseUrl = process.env.REACT_APP_API_URL;
+
+const SHORT_TEXT = "short-text";
+const SINGLE_CHOICE = "single-choice";
+const LONG_TEXT = ["long-text", "long_text"];
+const MULTI_CHOICE = "multi-choice";
+const FILE = "file";
+
+class RegistrationComponent extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            form: null,
+            error: "",
+            hasValidated: false,
+            isValid: false,
+            isSubmitting: false,
+            offer: [],
+            questionSections: [],
+            uploadPercentComplete: 0,
+            answers: [],
+            registrationId: false,
+            registrationFormId: false,
+            formSucess: false,
+            formFailure: false
+        }
+
+    }
+
+    resetPage() {
+        this.componentDidMount();
+        this.setState({
+            formSucess: false
+        });
+    }
+
+    getDescription = (question) => {
+        if (question.description) {
+            return question.description;
+        }
+    }
+
+    handleChange = event => {
+        const value = event.target.type === 'checkbox' ? (event.target.checked | 0) : event.target.value;
+        let answers = this.state.answers;
+        let id = parseInt(event.target.id);
+
+        let answer = answers.find(a => a.registration_question_id === id);
+        if (answer) {
+            answer.value = value.toString();
+            answers = answers.map(function (item) { return item.registration_question_id == id ? answer : item; });
+        }
+        else {
+            answer = {
+                registration_question_id: parseInt(id),
+                value: value.toString()
+            }
+            answers.push(answer);
+        }
+        this.setState({
+            answers: answers
+        }, );
+
+    };
+
+    handleChangeDropdown = (id, dropdown) => {
+        var answers = this.state.answers;
+
+        var answer = answers.find(a => a.registration_question_id === id);
+        if (answer) {
+            answer.value = dropdown.value.toString();
+            answers = answers.map(function (item) { return item.registration_question_id == id ? answer : item; });
+        }
+        else {
+            answer = {
+                registration_question_id: id.toString(),
```

Comment:
Howcome registration_question_id is a string here, but an int in handleChange? Since there's a lot of repeated functionality here, lets make a function to handle the common elements.. (optional)